# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/win/RemoteAccessible.cpp
# Commit: 3cfe5bd5d084
# Full Hash: 3cfe5bd5d084baf18f5d95b0e879f6fc179fac34
# Author: Morgan Reschenberg <mreschenberg@mozilla.com>
# Date: 2021-10-06 04:09:04
# Regressor Bug: 1726227
# File Overlap Count: 1
# Description:
#   Bug 1726227: Cache parent-relative accessible bounds, resolution in parent process r=Jamie,eeejay
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123399
# ==============================================================================

diff -r 76a965804210 -r 3cfe5bd5d084 accessible/ipc/win/RemoteAccessible.cpp
--- a/accessible/ipc/win/RemoteAccessible.cpp	Tue Oct 05 20:24:03 2021 +0000
+++ b/accessible/ipc/win/RemoteAccessible.cpp	Tue Oct 05 20:44:46 2021 +0000
@@ -219,7 +219,11 @@
   return state;
 }
 
-nsIntRect RemoteAccessible::Bounds() {
+nsIntRect RemoteAccessible::Bounds() const {
+  if (StaticPrefs::accessibility_cache_enabled_AtStartup()) {
+    return RemoteAccessibleBase<RemoteAccessible>::Bounds();
+  }
+
   nsIntRect rect;
 
   RefPtr<IAccessible> acc;